libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -394,6 +394,7 @@ func (k *kubernetesAnnotator) Close() error {
 	if k.cache != nil {
 		k.cache.stop()
 	}
+	k.initOnce.Do(func() {})
```

Comment:
Just to double-check, you're trying to avoid the situation where `Close` is called while `init` is still executing.
So, in that case, shouldn't this call be the 1st thing, to ensure:
 - close only runs after init has finished
 - close closes all resources init created?

Also, as it not so straight forward to get why it's calling `Do` with an empty function, I believe it would be good to add a comment here.